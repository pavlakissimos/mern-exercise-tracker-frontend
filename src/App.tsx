import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";

import Navbar from "./components/Navbar";
import ExercisesList from "./components/ExercisesList";
import EditExercise from "./components/EditExercise";
import CreateExercise from "./components/CreateExercise";
import CreateUser from "./components/CreateUser";

const App = () => {
  return (
    <div>
      <Router>
        <Navbar />
        <br />
        <div className="container mx-auto px-4 sm:px-0">
          <Route exact path="/" component={ExercisesList} />
          <Route exact path="/edit/:id" component={EditExercise} />
          <Route exact path="/create" component={CreateExercise} />
          <Route exact path="/user" component={CreateUser} />
        </div>
      </Router>
    </div>
  );
};

export default App;
