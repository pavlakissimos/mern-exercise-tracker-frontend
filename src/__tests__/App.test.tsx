import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import App from "../App";

afterEach(cleanup);

describe("App", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
