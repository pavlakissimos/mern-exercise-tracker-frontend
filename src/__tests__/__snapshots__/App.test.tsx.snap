// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="sm:flex justify-between bg-gray-900 py-3 px-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <a
          class="text-xl text-gray-500"
          href="/"
        >
          ExcerTrakr
        </a>
        <div
          class="sm:hidden"
        >
          <button
            class="sm:hidden text-gray-500 hover:text-white focus:text-white focus:outline-none"
            type="button"
          >
            <svg
              class="h-6 w-6 fill-current"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 5h16a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <nav
        class="hidden sm:flex text-gray-500 px-0 pt-2 sm:px-2 sm:pt-0 pb-4 sm:p-0"
      >
        <a
          class="block py-1 px-0 sm:px-2 font-semibold rounded hover:bg-gray-800"
          href="/"
        >
          Exercises
        </a>
        <a
          class="py-1 px-0 sm:px-2 font-semibold rounded hover:bg-gray-800 sm:mt-0 sm:ml-2"
          href="/create"
        >
          Create Exercise log
        </a>
        <a
          class="py-1 px-0 sm:px-2 font-semibold rounded hover:bg-gray-800 sm:mt-0 sm:ml-2"
          href="/user"
        >
          Create User
        </a>
      </nav>
    </header>
    <br />
    <div
      class="container mx-auto px-4 pb-6 sm:px-0"
    >
      <div
        class="grid xl:grid-cols-3 md:grid-cols-2 sm:grid-cols-1 grid-cols-1 gap-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
